src/lazyloading.js

```diff
@@ -1,53 +1,136 @@
-                }
-            });
-        };
+export default {
+  name: 'CoreLazyLoading',
+  props: {
+    placeholderImage: {
+      type: Object,
+      default: () => {
+        return imageDefault
+      },
+    },
+
+    options: {
+      type: Object,
+      default: () => {
+        return observerDefault
+      },
+    },
+  },
+  data() {
+    return {
+      observer: {},
+      firstStepToLoad: true,
+      image: {
+        class: '',
+        src: '',
+      },
+    }
+  },
+  render() {
+      return this.$slots.default;
+  },
+  mounted() {
+      this.setImageData();
 
-        const lazyImage = () => {
-            const options = {
-                root: null,
-                threshold: 1,
-            };
-            const observer = new IntersectionObserver(callback, options);
-            observer.observe(el);
-        };
+      this.$el.onload = this.onLoadImage;
+      this.$el.onerror = this.onErrorImage;
 
-        el.onload = () => {
-            if (
-                el.getAttribute('src') !==
-                'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAALwAAADGAQMAAABSPnhPAAAAA1BMVEXm5uSlVuPDAAAAG0lEQVR4Ae3BMQEAAADCIPunNsN+YAAAAABAdBNWAAHMZ56dAAAAAElFTkSuQmCC'
-            ) {
-                el.setAttribute('style', '');
-                el.setAttribute('class', className);
-            }
-        };
+      const observer = new IntersectionObserver(
+          this.checkIsIntersecting,
+          this.options
+      );
 
-        el.onerror = () => {
-            el.setAttribute('src', require('./assets/img/photo.png'));
-            el.setAttribute('style', '');
-            el.setAttribute('class', className);
-        };
+      observer.observe(this.$el);
+  },
+  methods: {
+    /**
+     * Start data to component
+     * @returns {null}
+     * */
+    setImageData() {
+          this.image = {
+              src: this.$el.getAttribute('src'),
+              class: this.$el.getAttribute('class'),
+          };
 
-        lazyImage();
+          this.changeElementAttribute(
+              ['class', 'style', 'src'],
+              [
+                  this.placeholderImage.class,
+                  this.placeholderImage.style,
+                  this.placeholderImage.src,
+              ]
+          );
     },
-};
+    /**
+     * Load image from finish observer
+     * @returns {null}
+     * */
+    loadImage() {
+        this.$el.src = this.image.src;
+        this.firstStepToLoad = false;
+    },
+    /**
+     * Event on load image
+     * @returns {null}
+     * */
+    onLoadImage() {
+      if (!this.firstStepToLoad) {
+          this.changeElementAttribute(['class', 'style'], [this.image.class, '']);
+      }
+    },
+    /**
+     * Event on error image
+     * @returns {null}
+     * */
+    onErrorImage() {
+        this.changeElementAttribute(
+            ['class', 'style', 'src'],
+            [this.image.class, '', require('./assets/img/photo.png')]
```

Comment:
`onErrorImage` - the image should be given as props. The user decides what picture should be given to the website, not we...

Reply:
ok bro I'll fix it